fut/src/App.js

```diff
@@ -3,11 +3,33 @@ import './App.css';
 import Topbar from './components/Topbar';
 
 class App extends Component {
+
+  constructor(props){
+    super(props);
+    this.state = {
+      input : '',
+    }
+    this.onInputChange = this.onInputChange.bind(this);
+  }
+  
+  componentDidMount(){
+    //fetch('http://localhost:8080')
+    fetch('https://api.sheety.co/32cddc1c-d9cb-4439-b995-6c204bc0e5c0')
+    .then(response => response.json())
+    .then(data => console.log(data[0].name))
+    .catch(error => console.log('error is', error));
+  }
+
+  onInputChange = (event) =>{
+    this.setState({input: event.target.value});
+    console.log(this.state.input);
+  }
```

Comment:
je sais pas si t'as remarqué mais la premiere lettre est pas loggé bizarement. En fait la maniere d'updater le state est correcte, mais il faut utiliser le callback de this.setState pour etre sur de choper le bon state un fois qu'une lettre est rentrée. 

`this.setState({ input: event.target.value }, () => console.log(this.state.input));`

si tu fais ca tu verra la premiere lettre loggée. La raison c'est que setState est async, donc le `this.state.input` que tu log juste en dessous de this.setState({input: event.target.value}); n'est pas encore le nouveau state. le callback passé en second argument a setState est appelé qu'une fois que le nouveau state est set donc tu est sur d'avoir le state le plus up to date dedans